src/usb_comm/usb_comm.c

```diff
@@ -19,38 +19,236 @@
  * this code. If not, see <http://www.gnu.org/licenses/>.
  */
 
-
 #include "FreeRTOS.h"
 #include "USB-CDC_device.h"
+#include "loggerApi.h"
+#include "loggerSampleData.h"
 #include "messaging.h"
+#include "panic.h"
+#include "printk.h"
+#include "rx_buff.h"
 #include "serial.h"
 #include "task.h"
 #include "usb_comm.h"
+#include <stdlib.h>
+#include <string.h>
 
+#define LOG_PFX			"[USB] "
+#define RX_BUFF_SIZE		1024
 #define USB_COMM_STACK_SIZE	320
-#define BUFFER_SIZE		1024
+#define USB_EVENT_QUEUE_DEPTH	8
+
+static volatile struct {
+        xQueueHandle event_queue;
+        struct Serial* serial;
+        struct {
+                struct rx_buff* buff;
+        } rx;
+} usb_state;
+
+
+static void log_event_overflow(const char* event_name)
+{
+        pr_warning_str_msg(LOG_PFX "Event overflow: ", event_name);
+}
+
+/**
+ * Structure used to house sample data for a wifi telemetry stream until
+ * the usb_event handler picks it up.
+ */
+struct usb_sample_data {
+        const struct sample* sample;
+        size_t tick;
+};
+
+/**
+ * Event struct used for all events that come into our wifi task
+ */
+struct usb_event {
+        enum task {
+                TASK_RX_DATA,
+                TASK_SAMPLE,
+        } task;
+        union {
```

Comment:
Yeah.  Leaving the door open to future expansion as needed without the need to go back and change it all.